src/NATS.Client.Core/NatsSub.cs

```diff
@@ -45,24 +42,25 @@ public sealed class NatsSub : NatsSubBase
         AllowSynchronousContinuations = false,
     });
 
+    internal NatsSub(NatsConnection connection, SubscriptionManager manager, string subject, string? queueGroup, int sid)
+        : base(connection, manager, subject, queueGroup, sid)
+    {
+    }
+
     public ChannelReader<NatsMsg> Msgs => _msgs.Reader;
 
-    public override async ValueTask DisposeAsync()
+    public override void Dispose()
```

Comment:
Should this remain `DisposeAsync` and wait for the UNSUB to complete?

Even if we don't await the UNSUB, I am a bit wary of using `Dispose` as opposed to `DisposeAsync` here should we ever want to await a task in the future.